### PR TITLE
Add user interaction lifecycle to interactive diagram core docs

### PR DESCRIPTION
**Pull request by michaellisitsa**
- [Add] Docs: Update execution lifecycle for interactive diagram to describe get/setStoredParams params

Further best practices for attaching event listeners to user interaction in future PR

### CODE REVIEW

File: docs/interactive-diagram-core.md [hunk 1]
```diff
@@ -4,9 +4,45 @@
...
+
+The [initialize](https://github.com/ClearCalcs/custom-diagram-boilerplate/blob/main/src/interactive/interface.ts#L7) function is called automatically once the first time the interactive diagram is loaded into a calculator.
+
+#### Use Cases
+
+-   Initialize third-party packages
+-   Data fetching e.g. GIS / CAD data
+-   any other async code that needs to run before `render` is called should be run here.
+-   Render initial state. This is useful where the diagram hasn't yet received any params e.g. builder's diagram widget.
+-   adding event listeners on user interaction e.g. click handlers.
```
[the remark this answers is not in the document]
Comment (michaellisitsa) [reply docs/interactive-diagram-core.md]: Fixed in c64f8cf5c274afea3518706a5bf43901078b0575

File: docs/interactive-diagram-core.md [hunk 2]
```diff
@@ -4,9 +4,45 @@
...
-1. **loading**: HTML/CSS loaded and Javascript executed (imported dependencies, top-level code)
-2. **initializing**: [initialize()](https://github.com/ClearCalcs/custom-diagram-boilerplate/blob/main/src/interactive/interface.ts#L1) executed, await return/resolve.
-3. **rendering**: [render(params)](https://github.com/ClearCalcs/custom-diagram-boilerplate/blob/main/src/interactive/interface.ts#L3) executed every time user changes params from sheet.
+### 1. Load
+
+The provided HTML markup and associated stylesheets are loaded into the DOM when a user opens a ClearCalcs calculator.
+
+### 2. Initialize
+
+The [initialize](https://github.com/ClearCalcs/custom-diagram-boilerplate/blob/main/src/interactive/interface.ts#L7) function is called automatically once the first time the interactive diagram is loaded into a calculator.
```
[the remark this answers is not in the document]
Comment (michaellisitsa) [reply docs/interactive-diagram-core.md]: Fixed in c64f8cf5c274afea3518706a5bf43901078b0575

File: docs/interactive-diagram-core.md [hunk 3]
```diff
@@ -4,9 +4,45 @@
```
[the remark this answers is not in the document]
Comment (michaellisitsa) [reply docs/interactive-diagram-core.md:4]: Usually there should be a `localhost:4444` link somewhere. The script uses `npm` so maybe an incompatibility there with `pnpm`

**Comment by michaellisitsa** [comment]
@JulianGoh5 Ready for re-review

**Review by JulianGoh5 — APPROVED** [review]

**Pull request merged**